File: program48.c
///////////////////////////////////////////////////////
//
//  File name :     program48.c
//  Descreption :   Separate Digit and Count Digits from Number
//  Author :        Vaibhav
//  Date :          12/05/2025
//
///////////////////////////////////////////////////////

#include<stdio.h>

int CountDigits(int iNo)
{
    int iCnt = 0;

    while(iNo != 0)
    {
        iNo = iNo / 10;
        iCnt++;
    }
    return iCnt;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    printf("Enter number: \n");
    scanf("%d",&iValue);

    printf("-----------------------------------------------------------------------------------------------\n");
    printf("Number Received is %d\n",iValue);
    printf("-----------------------------------------------------------------------------------------------\n");


    iRet = CountDigits(iValue);

    printf("Number of Digits are : %d\n",iRet);
    
    return 0;
}

File: Java/program48.java
///////////////////////////////////////////////////////
//
//  File name :     program48.java
//  Descreption :   Separate Digit and Count Digits from Input Number
//  Author :        Vaibhav
//  Date :          20/05/2025
//
///////////////////////////////////////////////////////

import java.util.Scanner;

class CountDigits
{
    int iValue;

    int Display()
    {
        int iCnt = 0;
        
        while(iValue != 0)
        {
            iValue = iValue / 10;
            iCnt++;
        }
        return iCnt;
    }
}   
class program48
{
    public static void main(String A[])
    {
        int iNo = 0;
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number : ");
        iNo = sobj.nextInt();

        System.out.println("-----------------------------------------------------------------------------------------------");
        System.out.println("Number Received is "+iNo);
        System.out.println("-----------------------------------------------------------------------------------------------");
        

        CountDigits dobj = new CountDigits();

        dobj.iValue = iNo;

        iRet = dobj.Display();

        System.out.println("Number of Digits in Number "+iNo+" is "+iRet);

    }
}